docs/proposals/service-catalog.md

```diff
@@ -0,0 +1,143 @@
+
+
+## Custom dynamic provisioning
+
+The IT department manages an off-cluster database. Each developer wanting to access the database is required to use a unique username and password to access the database. Additionally, each developer accesses a unique tablespace that no other team members can access. The IT department used to create database accounts and tablespaces by hand in response to individual developer requests.
+
+Moving forward, IT wants to automate the process. They create a web service that implements the "service broker" API (such as the one from [Cloud Foundry](http://docs.cloudfoundry.org/services/api.html#api-overview)). They add an entry to the service catalog for their database service, pointing at their web service. When a developer consumes the entry from the catalog, IT's web service is contacted, resulting in a new username, password, and tablespace.
+
+# Service configuration data
+
+The only data currently associated with a service is a list of zero or more endpoints. Many services also have configuration data associated with them. These include things such as logins, passwords, and additional connection parameters. To tie configuration data to a service, we propose adding information to the Service type to be able to express a relationship between the Service and its relevant configuration data. This could be accomplished by adding new field(s) or by using annotations. Potential configuration data target reference types include Secrets and ConfigData.
+
+Just creating the reference association by itself does not accomplish much; to be useful, that data needs to be available to the processes executing in containers. More on this process is below in the sections on service claims and linking.
+
+# Service catalogs
+
+A service catalog is a listing of "services". Examples might include an external database that is deployed outside of the cluster, an actual Kubernetes Service, a template of cluster primitives to be created, and custom provisioners that can perform tasks such as creating users in external systems.
+
+The catalog is not meant to include every service in the cluster. Instead, it should contain those services that users wish to highlight and make available to other users. For example, your namespace might contain "etcd", "etcd-discovery", and "postgresql" services, and the only one you want to share with others is the postgresql service.
+
+## Publishing to a catalog
+
+Users should be able to publish entries to a service catalog. One means of accomplishing could be by adding annotations to the resources that can go into a catalog (e.g, Services). Another option would be to create `ServiceCatalogEntry` resources that reference the appropriate resources.
```

Comment:
How does publishing/viewing work with security for the initial iteration?  Is the intention to let the claim processing handle rejecting invalid requests rather than limiting visibility to only things you can claim?